Sources/Datadog/FeaturesIntegration/WebView/WebRUMEventConsumer.swift

```diff
@@ -6,8 +6,92 @@
+        let eventData = try JSONSerialization.data(withJSONObject: event, options: [])
+        let jsonDecoder = JSONDecoder()
+        let rumContext = contextProvider?.context
+
+        switch eventType {
+        case "view":
+            let viewEvent = try jsonDecoder.decode(RUMViewEvent.self, from: eventData)
+            let mappedViewEvent = mapIfNeeded(dataModel: viewEvent, context: rumContext, offset: getOffset(viewID: viewEvent.view.id))
+            write(mappedViewEvent)
+        case "action":
+            let actionEvent = try jsonDecoder.decode(RUMViewEvent.self, from: eventData)
+            let mappedActionEvent = mapIfNeeded(dataModel: actionEvent, context: rumContext, offset: getOffset(viewID: actionEvent.view.id))
+            write(mappedActionEvent)
+        case "resource":
+            let resourceEvent = try jsonDecoder.decode(RUMViewEvent.self, from: eventData)
+            let mappedResourceEvent = mapIfNeeded(dataModel: resourceEvent, context: rumContext, offset: getOffset(viewID: resourceEvent.view.id))
+            write(mappedResourceEvent)
+        case "error":
+            let errorEvent = try jsonDecoder.decode(RUMViewEvent.self, from: eventData)
+            let mappedErrorEvent = mapIfNeeded(dataModel: errorEvent, context: rumContext, offset: getOffset(viewID: errorEvent.view.id))
+            write(mappedErrorEvent)
+        case "long_task":
+            let longTaskEvent = try jsonDecoder.decode(RUMViewEvent.self, from: eventData)
+            let mappedLongTaskEvent = mapIfNeeded(dataModel: longTaskEvent, context: rumContext, offset: getOffset(viewID: longTaskEvent.view.id))
+            write(mappedLongTaskEvent)
+        default:
+            userLogger.error("🔥 Web RUM Event Error - Unknown event type: \(eventType)")
+        }
+    }
+
+    private func mapIfNeeded<T: RUMDataModel>(dataModel: T, context: RUMContext?, offset: Offset) -> T {
+        guard let context = context else {
+            return dataModel
+        }
+        // TODO: RUMM-1786 implement mappers
+        let mappedDataModel = dataModel
+        return mappedDataModel
+    }
+
+    private func write<T: RUMDataModel>(_ model: T) {
+        dataWriter?.write(value: model)
+    }
+
+    // MARK: - Time offsets
+
+    private typealias Offset = TimeInterval
+    private typealias ViewIDOffsetPair = (viewID: String, offset: Offset)
+    private var viewIDOffsetPairs = [ViewIDOffsetPair]()
+
+    private func getOffset(viewID: String) -> Offset {
+        purgeOffsets()
+
+        let found = viewIDOffsetPairs.first { $0.viewID == viewID }
+        if let found = found {
+            return found.offset
+        }
+        let offset = dateCorrector?.currentCorrection.serverTimeOffset ?? 0.0
+        viewIDOffsetPairs.insert((viewID: viewID, offset: offset), at: 0)
+        return offset
+    }
+
+    private func purgeOffsets() {
+        while viewIDOffsetPairs.count > 3 {
+            _ = viewIDOffsetPairs.popLast()
+        }
```

Comment:
What are you solving with this cache of offsets? 🤔 The server time offset can be considered as constant, and if it's not (e.g. during sync phase) the last offset should be use.

Reply:
i did ask almost the same question :) 
this was discussed here: https://github.com/DataDog/dd-sdk-ios/pull/683#discussion_r765099800

Reply:
Where is the discussion?

Reply:
Ok, I get it! we need all events to be in synced with the parent view event.
Note: the sync phase of Kronos is too long IMO, that's something we should revisit!